src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToGenerateImplicitNamespaceImports_DotNet.cs

```diff
@@ -0,0 +1,216 @@
+@"// <autogenerated />
+global using global::CustomNamespace;
+");
+        }
+
+        [Fact]
+        public void It_ignores_duplicate_imports()
+        {
+            var tfm = "net6.0";
+            var testProject = CreateTestProject(tfm);
+            testProject.AdditionalProperties["DisableImplicitNamespaceImports_DotNet"] = "true";
+            testProject.AdditionalItems["Import"] = new Dictionary<string, string> { ["Include"] = "CustomNamespace;CustomNamespace" };
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var importFileName = $"{testAsset.TestProject.Name}.ImplicitNamespaceImports.cs";
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand
+                .Execute()
+                .Should()
+                .Fail();
+
+            var outputDirectory = buildCommand.GetIntermediateDirectory(tfm);
+
+            outputDirectory.Should().HaveFile(importFileName);
+
+            File.ReadAllText(Path.Combine(outputDirectory.FullName, importFileName)).Should().Be(
+@"// <autogenerated />
+global using global::CustomNamespace;
+");
+        }
+
+        [Fact]
+        public void It_can_disable_import_generation()
+        {
+            var tfm = "net6.0";
+            var testProject = CreateTestProject(tfm);
+            testProject.AdditionalProperties["DisableImplicitNamespaceImports"] = "true";
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var importFileName = $"{testAsset.TestProject.Name}.ImplicitNamespaceImports.cs";
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand
+                .Execute()
+                .Should()
+                .Fail();
+
+            var outputDirectory = buildCommand.GetIntermediateDirectory(tfm);
+
+            outputDirectory.Should().NotHaveFile(importFileName);
+        }
+
+        [Fact]
+        public void It_ignores_invalid_tfms()
+        {
+            var tfm = "net5.0";
+            var testProject = CreateTestProject(tfm);
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var importFileName = $"{testAsset.TestProject.Name}.ImplicitNamespaceImports.cs";
+
+            var buildCommand = new BuildCommand(testAsset);
+            buildCommand
+                .Execute()
+                .Should()
+                .Fail();
+
+            var outputDirectory = buildCommand.GetIntermediateDirectory(tfm);
+
+            outputDirectory.Should().NotHaveFile(importFileName);
+        }
+
+        private TestProject CreateTestProject(string tfm)
+        {
+            var testProject = new TestProject
+            {
+                IsExe = true,
+                TargetFrameworks = tfm,
+                ProjectSdk = "Microsoft.NET.Sdk"
+            };
+            testProject.AdditionalProperties["LangVersion"] = "10.0";
```

Comment:
Simulates M.CSharp.Core.Targets: https://github.com/dotnet/roslyn/blob/9e30164a1ea97f6f18d48743b05ec68f09fba301/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets#L20-L22

Reply:
If you update stage 0, you shouldn't need this.  You can cherry-pick this commit: a07b803919b8750f461bf7cadf3a7508e64db978